lib/iiif_print/persistence_layer/valkyrie_adapter.rb

```diff
@@ -68,10 +68,24 @@ def self.solr_name(field_name)
         Hyrax.config.index_field_mapper.solr_name(field_name.to_s)
       end
 
-      ##
-      # @todo implement this logic
-      def self.destroy_children_split_from(file_set:, work:, model:)
-        super
+      def self.destroy_children_split_from(file_set:, work:, model:, user:)
+        # look first for children by the file set id they were split from
+        children = Hyrax.query_service.find_all_of_model(model: model).select { |m| m.split_from_pdf_id == file_set.id }
```

Comment:
Yikes.  This is wildly non-performant.  We'll need to look for another approach.  What's the given `model`?  We could end up looping through millions of records to find these elements.

I suspect we're into writing custom query logic; or reviewing other options.

Reply:
I think we are looking to use `.find_inverse_references_by`

```
Hyrax.query_service.find_inverse_references_by(resource: file_set, model: model, property: "split_from_pdf_id").each do |child|
```

See implementation details:  https://github.com/samvera/valkyrie/blob/00077962af3acc444657c0d664a7fa0f282c4a52/lib/valkyrie/persistence/postgres/query_service.rb#L116-L126